packages/backend-common/src/tokens/ServerTokenManager.ts

```diff
@@ -56,44 +56,60 @@ export class ServerTokenManager implements TokenManager {
         'You must configure at least one key in backend.auth.keys for production.',
       );
     }
-
     // For development, if a secret has not been configured, we auto generate a secret instead of throwing.
-    const generatedDevOnlyKey = JWK.generateSync('oct', 24 * 8);
-    if (generatedDevOnlyKey.k === undefined) {
-      throw new Error('Internal error, JWK key generation returned no data');
-    }
     logger.warn(
       'Generated a secret for backend-to-backend authentication: DEVELOPMENT USE ONLY.',
     );
-    return new ServerTokenManager([generatedDevOnlyKey.k]);
+    return new ServerTokenManager([]);
   }
 
   private constructor(secrets: string[]) {
-    if (!secrets.length) {
+    if (!secrets.length && process.env.NODE_ENV !== 'development') {
       throw new Error(
         'No secrets provided when constructing ServerTokenManager',
       );
     }
+    this.verificationKeys = secrets.map(s => base64url.decode(s));
+    this.signingKey = this.verificationKeys[0];
+  }
 
-    this.verificationKeys = new JWKS.KeyStore(
-      secrets.map(k => JWK.asKey({ kty: 'oct', k })),
-    );
-    this.signingKey = this.verificationKeys.all()[0];
+  // Called when no keys have been generated yet in the dev environment
+  private async generateKeys(): Promise<void> {
+    if (process.env.NODE_ENV !== 'development') {
+      throw new Error(
+        'Key generation is not supported outside of the dev environment',
+      );
+    }
+    const secret = await generateSecret('HS256');
+    const jwk = await exportJWK(secret);
+    this.verificationKeys.push(base64url.decode(jwk.k ?? ''));
+    this.signingKey = this.verificationKeys[0];
   }
 
   async getToken(): Promise<{ token: string }> {
-    const jwt = JWT.sign({ sub: 'backstage-server' }, this.signingKey, {
-      algorithm: 'HS256',
-    });
-
+    if (!this.verificationKeys.length) {
+      await this.generateKeys();
```

Comment:
There's a bit of a race here that I don't think will cause any trouble, but perhaps it's best to fix it anyway. If there are multiple callers of `getToken` we may end up generating multiple keys.

Reply:
Yep, something like what the `TokenFactory` does

Reply:
Updated the implementation here to match `TokenFactory`